### PR TITLE
markdown description

### PR DESCRIPTION
**Pull request by chloeleichen**
Fixing https://github.com/TerriaJS/magda/issues/229, ~probably~ definitely in an overly complicated way, happy to hear what you think.

### CODE REVIEW

**Comment by kring** [comment]
@chloeleichen you should use the same approach we use in TerriaJS.  The one here is potentially vulnerable  to XSS attacks and that sort of thing.  Remember that markdown can contain HTML, and markdown-it doesn't sanitize it, so using `dangerouslySetInnerHTML` is ... dangerous!

File: magda-web-client/src/UI/MarkdownViewer.js [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
+  constructor(props) {
+   super(props);
+   this.state = {isCollapsed: true};
+ }
 
-function MarkdownViewer(props) {
-    let markdown = {__html: marked(props.markdown)};
-    return <div className='markdown' dangerouslySetInnerHTML={markdown}/>
+ componentDidMount(){
+   const numberOfChildrenFromMarkdownAsHtml = this.markdown.children.length;
```
Comment (kring) [inline magda-web-client/src/UI/MarkdownViewer.js]: This should probably happen more often than just the initial mount, right?  Like if the markdown changes and the content length increases, a user of `MarkdownViewer` would probably expect `updateContentLength` to be called again.

**Comment by chloeleichen** [comment]
updated, @kring can you have a look again.

**Comment by kring** [comment]
@chloe do you remember why we ended up using `html-to-react` rather than `dangerouslySetInnerHTML` in TerriaJS?  I'm just curious.  This should be ok from an XSS perspective now, at least.

**Comment by kring** [comment]
Please merge in master (there are conflicts) and then I'll merge this.

**Comment by chloeleichen** [comment]
I'm going to wait for https://github.com/TerriaJS/magda/pull/235 to merge first as there are some directory changes in there

**Pull request merged**